algosdk/future/transaction.py

```diff
@@ -2533,7 +2533,7 @@ def verify(self, public_key):
             try:
                 verify_key.verify(to_sign, base64.b64decode(self.sig))
                 return True
-            except BadSignatureError:
+            except (BadSignatureError, ValueError):
```

Comment:
Note: There was [a change in pynacl](https://github.com/pyca/pynacl/pull/600/files) that allows throwing a `ValueError` or `TypeError` in the `verify` function.